src/jackdaw/serdes.clj

```diff
@@ -1,8 +1,72 @@
 (ns jackdaw.serdes
-  "Implements basic SerDes (Serializer/Deserializer)."
+  "Implements string and EDN serdes (serializer/deserializer)."
   (:gen-class)
-  (:import [org.apache.kafka.common.serialization Serdes]))
+  (:require [clojure.edn]
+            [jackdaw.serdes.fn :as jsfn])
+  (:import java.nio.charset.StandardCharsets
+           org.apache.kafka.common.serialization.Serde
+           org.apache.kafka.common.serialization.Serdes))
+
+
+(set! *warn-on-reflection* true)
+
+
+(defn to-bytes
+  "Converts a string to a byte array."
+  [data]
+  (.getBytes ^String data StandardCharsets/UTF_8))
+
+(defn- from-bytes
+  "Converts a byte array to a string."
+  [^bytes data]
+  (String. data StandardCharsets/UTF_8))
+
 
 (defn string-serde
   []
   (Serdes/String))
+
+
+(defn edn-serializer
+  "Returns an EDN serializer."
+  []
+  (jsfn/new-serializer {:serialize (fn [_ _ data]
+                                     (when data
+                                       (to-bytes
+                                        (binding [*print-length* false
+                                                  *print-level* false]
+                                          (pr-str data)))))}))
+
+(defn edn-deserializer
+  "Returns an EDN deserializer."
+  ([]
+   (edn-deserializer {}))
+  ([opts]
+   (let [opts (into {} opts)]
+     (jsfn/new-deserializer {:deserialize (fn [_ _ data]
+                                            (->> (from-bytes data)
+                                                 (clojure.edn/read-string opts)))}))))
+
+(defn edn-serde
```

Comment:
What do you think about making it so that just this bit (i.e. only the function which returns the serde) is in the `jackdaw.serdes` ns and the interop stuff is in some other namespace (suggest jackdaw.serdes.edn2 but not too fussed about the exact name).